### PR TITLE
Update sbt-header plugin and copyright headers

### PR DESCRIPTION
**Pull request by richdougherty**
This updates us to the latest version of the sbt-header plugin, a version which supports sbt 1. Since this version of the sbt-header plugin changed the project headers I decided to just update all the headers to 2018. I also added header checks to the sources in the documentation.

I hope you enjoy reviewing this PR. :)

### CODE REVIEW

File: documentation/build.sbt [hunk 1]
```diff
@@ -46,7 +46,9 @@ lazy val main = Project("Play-Documentation", file(".")).enablePlugins(PlayDocsP
       scalaVersion := PlayVersion.scalaVersion,
 
       fork in Test := true,
-      javaOptions in Test ++= Seq("-Xmx512m", "-Xms128m")
+      javaOptions in Test ++= Seq("-Xmx512m", "-Xms128m"),
+
+      headerLicense := Some(HeaderLicense.Custom("Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>"))
```
Comment (marcospereira) [inline documentation/build.sbt]: 💯

**Comment by richdougherty** [comment]
Now use `headerCheck` command. And yes I've already got it checking the doc headers.

Note there are a lot of files that we're still not automatically checking for headers - Markdown, configuration files, etc. I didn't want to extend this PR any further than it already is by trying to check new types of files with new comment formats.

**Comment by mkurz** [comment]
@richdougherty CI fails:
```
[error] (Play-Akka-Http-Server/test:headerCheck) There are files without headers!
[error]   /home/travis/build/playframework/playframework/framework/src/play-akka-http-server/src/test/scala/play/core/server/akkahttp/AkkaHeadersWrapperTest.scala
```

**Comment by schmitch** [comment]
Also the plugins adds a white line between the header and the package, is this wished/planned/required?

before:
```
/*
 * Copyright (C) 2009-2017 Lightbend Inc. <https://www.lightbend.com>
 */
package javaguide.detailed.filters.removeAll;
```

after:
```
/*
 * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
 */

package javaguide.detailed.filters.removeAll;
```

**Comment by mkurz** [comment]
@schmitch We are aware about that. See https://github.com/sbt/sbt-header/pull/117#issuecomment-367388250

**Comment by raboof** [comment]
Rebased, updated and merged in #8380, so I think we can close this one

**Comment by richdougherty** [comment]
Thanks @raboof!

**Pull request closed without merging**